### PR TITLE
Add fluent resources to logging

### PR DESCRIPTION
**Pull request by nickgerace**
Add fluentbit and fluentd resources to logging. Add clearer comments on
the two psp settings.

Relevant issue: https://github.com/rancher/rancher/issues/31099

### CODE REVIEW

**Comment by nickgerace** [comment]
This is pending merge of https://github.com/rancher/charts/pull/963 and rebase afterwords (new RC version).

**Review by paynejacob — APPROVED** [review]

**Review by brendarearden — APPROVED** [review]
LGTM

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch [hunk 1]
```diff
@@ -26,16 +26,17 @@
...
  
-@@ -50,6 +55,8 @@
+@@ -50,6 +55,9 @@
      # Labels to query http service
      labels: {}
  
 +# These "rbac" settings match the upstream defaults. For only using psp in the overlay files, which
-+# include the default Logging CRs created, see the "global.psp" setting.
++# include the default Logging CRs created, see the "global.psp" setting. To enable psp for the entire
++# chart, enable both "rbac.psp" and "global.psp" (this may require further changes to the chart).
```
Comment (cbron) [inline packages/rancher-logging/generated-changes/patch/values.yaml.patch:35]: Wasn't this from other PR ? We can include here if its just cleanup. Want to make sure we aren't seeing a conflict.

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml [hunk 2]
```diff
@@ -28,6 +28,10 @@ spec:
     nodeSelector:
       {{- toYaml . | nindent 6 }}
   {{- end }}
+  {{- with .Values.fluentbit.resources }}
```
Comment (cbron) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:31]: Where is fluentbit.resources set ? I don't see it in values.yaml

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch [hunk 3]
```diff
@@ -85,11 +86,15 @@
 +    repository: rancher/banzaicloud-fluentd
 +    tag: v1.11.5-alpine-9
 +
++fluentd:
++  resources: {}
```
Comment (cbron) [inline packages/rancher-logging/generated-changes/patch/values.yaml.patch:90]: How does this affect the chart level one ? Which takes precedence: https://github.com/rancher/charts/blob/dev-v2.5-alt/charts/rancher-logging/rancher-logging/3.9.000-rc03/values.yaml#L23

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch (hunk 1 above)
Comment (nickgerace) [reply packages/rancher-logging/generated-changes/patch/values.yaml.patch:35]: This is purposeful. I added more comments.
> Add fluentbit and fluentd resources to logging. Add clearer comments on
the two psp settings.

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml (hunk 2 above)
Comment (nickgerace) [reply packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:31]: Ah that was an oversight. Good catch.

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch (hunk 3 above)
Comment (nickgerace) [reply packages/rancher-logging/generated-changes/patch/values.yaml.patch:90]: The chart level is for the operator install. The fluentbit and fluentd settings are for the Logging CRs in the overlay.

I think this is out of scope for this PR, but we may want to reorganize values.yaml to have an "overlay" section (or something similar) to denote all of our additions.
CC: @paynejacob

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml (hunk 2 above)
Comment (nickgerace) [reply packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:31]: Oh wait no, it's there. Line 97.

Comment (nickgerace) [reply packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:31]: I moved them together for readability.

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch (hunk 3 above)
Comment (cbron) [reply packages/rancher-logging/generated-changes/patch/values.yaml.patch:94]: Ok. So to clarify, the chart level one did not affect fluent d/bit before and still won’t.

**Review by cbron — APPROVED** [review]

**Pull request merged**